### PR TITLE
Remote access v2

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25698

## Description
Support Remote Access integration from content side

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
       - Some of the parameters have been removed/some arguments names have been changed. Please let me know if you still want me to support old command argument names @dantavori or not. Detailed changes is found in the end of the README file
   - [ ] No

## Must have
- [x] Tests - Did not create a TPB for this as it might be problematic to run in build - if you feel it is required I can investigate further how to conduct a TPB for this integration (Need to configure some Host that can be accessed from the docker running the RemoteAccess integration, including some user credentials in the given Host) @dantavori 
- [x] Documentation

### CODE REVIEW

File: Packs/RemoteAccess/Integrations/RemoteAccessV2/RemoteAccessV2.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      description: The file type, as determined by libmagic (same as displayed in
+        file entries).
+      type: String
+  isfetch: false
+  runonce: false
+  script: '-'
+  type: python
+  subtype: python3
+  dockerimage: demisto/netmiko:1.0.0.15047
+fromversion: 5.5.0
```
Comment (Itay4) [inline Packs/RemoteAccess/Integrations/RemoteAccessV2/RemoteAccessV2.yml:124]:
```suggestion
fromversion: 6.0.0
```

File: Packs/RemoteAccess/Integrations/RemoteAccessV2/Pipfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+pytest-mock = "*"
+requests-mock = "*"
+pytest-asyncio = "*"
+
+[packages]
+pytest = "*"
+requests = "*"
+
+[requires]
+python_version = "3.7"
```
Comment (Itay4) [inline Packs/RemoteAccess/Integrations/RemoteAccessV2/Pipfile:18]: shouldnt this be `3.9` ?

**Comment by bakatzir** [comment]
relates to: https://github.com/demisto/etc/issues/32192

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    password: str = credentials.get('password', '')
+
+    host_name: str = params.get('hostname', '')
+
+    ciphers: Set[str] = set(argToList(params.get('ciphers')))
+    key_algorithms: Set[str] = set(argToList(params.get('key_algorithms')))
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    if password_param := params.get('additional_password'):
+        if password_param != args.get('additional_password'):
```
Comment (dantavori) [inline Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:213]: how does it work in v1?

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| RemoteAccess.Command.output | String | STD output of the given command. | 
+| RemoteAccess.Command.error | String | STD error output of the given command. | 
+| RemoteAccess.Command.success | Boolean | Whether operation have been successful. | 
+| RemoteAccess.Command.command | String | Command that have been ran. | 
+
+
+#### Command Example
+```!remote-access-ssh command=`echo test````
```
Comment (dantavori) [inline Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md:53]:
```suggestion
```!remote-access-ssh command="echo test"```
```

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+## RemoteAccess v2
+
+This integration enables Cortex XSOAR to access and run commands on a terminal in a remote location (via SSH).
+
+### Ciphers
+- When ciphers are given, the SSH connection will be established with the first cipher of the given ciphers that given host supports.
+
+### Key Algorithms
+- When key algorithms are given, the SSH connection will be established with the first key algorithm of the given key algorithms that given host supports.
```
Comment (dantavori) [inline Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md:9]: what is the default behavior if they are not given?

**Review by dantavori — CHANGES_REQUESTED** [review]
looks great
let's talk about the TPB today

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md (hunk 5 above)
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md:9]: It picks some Key/Cypher that exists. TBH not sure how it decides to pick one, can try and investigate

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+        copy_to_remote (bool): Whether a request to copy to remote was made.
+
+    Returns:
+        (str): Empty str if command requested was copy to.
+        (str): Str containing the copied file from remote machine data.
+    Raises:
+        (Exception):  if unexpected behaviour occurred.
+    """
+    try:
+        with SCPClient(ssh_client.get_transport()) as scp_client:
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:37]: Sounds like a good idea. How will we make sure it wont confuse the customer? e.g, setting 10m timeout but the command itself has a 5m timeout. Do you think adding the maximum value is 5m should be the best option?

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+| Remote Access Command Name | Old Command Argument Name | Remote Access v2 Command Argument Name | New Command Name |
+| --- | --- | --- | --- |
+| ssh | cmd | remote-access-ssh | command |
+| ssh | timeout | remote-access-ssh | **Argument have been removed** |
+| ssh | system | remote-access-ssh | **Argument have been removed** |
+| copy-to | dest-dir | remote-access-copy-to | destination_path |
+| copy-to | entry | remote-access-copy-to | entry_id |
+| copy-to | timeout | remote-access-copy-to | **Argument have been removed** |
+| copy-to | system | remote-access-copy-to | **Argument have been removed** |
+| copy-to | fileID | remote-access-copy-to | **Argument have been removed** |
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md:188]: No idea what was the fileID usage, i figured entry id is enough for this case, i can see how it is used

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py (hunk 3 above)
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:213]: you can configure additional password in the instance, and somehow you have to type that password in a command to be able to run the command

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:37]: It's possible to increase the default command timeout value in the server configurations. Also the default socket_timeout is 10m so if not given, the final timeout will be 5m (the command's timeout).

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py (hunk 3 above)
Comment (dantavori) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:213]: OK, let's set a quick call todat to talk about this and the TPB

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md (hunk 5 above)
Comment (dantavori) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2_description.md:9]: Please check and document

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/RemoteAccessv2.py:37]: I didnt understand what you want the behaviour to be
do you want me to change default timeout in server? or leave it as 5m? document the maximum socket timeout argument? i

**Comment by ShirleyDenkberg** [comment]
@Itay4 @dantavori Doc review completed.

**Review by bakatzir — COMMENTED** [review]
1. please put `additional password` as `credentials` as well - https://github.com/demisto/etc/issues/39286
2. should `remote-access-ssh` be `execution: true`? as this is a potentially harmful command. check what happens in v1.
3. add a note in `?`/readme faq what can be done in setup on a few setup that you already have. TBD with docs team.

**Comment by tomneeman151293** [comment]
@dantavori ready for re-review

**Review by dantavori — APPROVED** [review]

File: Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    | Password |  | False |
+    | Additional Password | Will require an additional password as an argument to run any command of this module. | False |
+    | Ciphers | A comma-separated list of ciphers to use. If none of the specified ciphers are agreed to by the server, an error message specifying the supported ciphers is returned. | False |
+    | Key Algorithms | A comma-separated list of key algorithms to use. If none of the specified key algorithms are agreed to by the server, an error specifying the supported key algorithms is returned. | False |
+
+4. Click **Test** to validate the URLs, token, and connection.
+
+## Config SSH From Remote
+For login using root:
+1. Edit /etc/ssh/sshd_config file
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply Packs/RemoteAccess/Integrations/RemoteAccessv2/README.md:25]: Can't SSH to windows atm

**Pull request merged**